lib/roken/issuid.c

```diff
@@ -62,23 +62,84 @@
 #endif
 #endif
 
-#ifdef HAVE_GETAUXVAL
-static unsigned long
+ROKEN_LIB_FUNCTION unsigned long ROKEN_LIB_CALL
+rk_getprocauxval(unsigned long type)
+{
+    static int has_proc_auxv = 1;
+    unsigned long a[2];
+    ssize_t bytes;
+    int save_errno = errno;
+    int fd;
+
+    if (!has_proc_auxv) {
+        errno = ENOENT;
+        return 0;
+    }
+
+    if ((fd = open("/proc/self/auxv", O_RDONLY)) == -1) {
+        if (errno == ENOENT)
+            has_proc_auxv = 0;
+        errno = ENOENT;
+        return 0;
+    }
+
+    do {
+        if ((bytes = read(fd, a, sizeof(a))) != sizeof(a))
+            break;
+        if (a[0] == type) {
+            (void) close(fd);
+            errno = save_errno;
+            return a[1];
+        }
+    } while (bytes == sizeof(a) && (a[0] != 0 || a[1] != 0));
+
+    (void) close(fd);
+    errno = ENOENT;
+    return 0;
+}
+
+ROKEN_LIB_FUNCTION unsigned long ROKEN_LIB_CALL
 rk_getauxval(unsigned long type)
 {
-    errno = 0;
+#ifdef HAVE_GETAUXVAL
 #ifdef GETAUXVAL_SETS_ERRNO
     return getauxval(type);
 #else
-    unsigned long ret = getauxval(type);
+    unsigned long ret;
+    unsigned long ret2;
+    static int getauxval_sets_errno = -1;
+    int save_errno = errno;
+    int fd;
```

Comment:
Correct.  I need to find a way to make this compile always so I can test it, no?